### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by fouadkachlan**

### CODE REVIEW

File: Components/App.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+  const completeTask = (taskNameToDelete: string): void => {
+    setTodoList(todoList.filter((task) => {
+      return task.taskName !== taskNameToDelete;
+    }));
+  };
+
+  return (
+    <CustomDiv>
+      <CustomDiv style={{
```
Comment (jihad292) [inline Components/App.tsx]: as we have seen we have always common styles that should be given to custom div like flex direction and alignItems and justify contents .. so lets set these styles inside Custom div and pass there values as props

File: Components/CustomDiv/CustomDiv.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+interface CustomeDivProps {
+  children: React.ReactNode;
+  style?: React.CSSProperties;
```
Comment (jihad292) [inline Components/CustomDiv/CustomDiv.tsx]: lets try no to use optional

File: Components/App.tsx (hunk 1 above)
Comment (fouadkachlan) [reply Components/App.tsx]: Done

File: Components/CustomDiv/CustomDiv.tsx (hunk 2 above)
Comment (fouadkachlan) [reply Components/CustomDiv/CustomDiv.tsx]: Done

File: Components/CustomDiv/CustomDiv.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+interface CustomeDivProps {
+  children: React.ReactNode;
+  style : React.CSSProperties;
+}
+
+const CustomeDiv: React.FC<CustomeDivProps> = ({ children, style }) => {
```
Comment (jihad292) [inline Components/CustomDiv/CustomDiv.tsx]: CustomDiv not CustomeDiv

File: Components/App.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    }));
+  };
+  const sameStyles = {
+    display:'flex',
+    justifyContent: 'center',
+    alignItems: 'center'
+  }
+
+  return (
+    <CustomDiv style={{ width:'110%'}}>
```
Comment (jihad292) [inline Components/App.tsx]: 110% ?

File: Components/App.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        width: '100vw',
+        height: '100vh',
+        fontFamily: 'Arial, Helvetica, sans-serif'
+      }}>
+        <CustomDiv style={{
+          ...sameStyles,
+          flex: 30,
+          backgroundColor: 'tomato',
+          width: '100%',
+          marginRight: '100px',
```
Comment (jihad292) [inline Components/App.tsx]: marginRight in px ?  and width and height in percentage? make all the unit same

File: Components/App.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+      <CustomDiv style={{
+        display: 'flex',
+        alignItems: 'center',
+        flexDirection: 'column',
+        width: '100vw',
+        height: '100vh',
+        fontFamily: 'Arial, Helvetica, sans-serif'
+      }}>
+        <CustomDiv style={{
+          ...sameStyles,
```
Comment (jihad292) [inline Components/App.tsx]: this modification should be inside CustomDiv only

File: Components/CustomDiv/CustomDiv.tsx (hunk 3 above)
Comment (fouadkachlan) [reply Components/CustomDiv/CustomDiv.tsx]: Done

Comment (fouadkachlan) [reply Components/CustomDiv/CustomDiv.tsx]: Done

File: Components/App.tsx (hunk 5 above)
Comment (fouadkachlan) [reply Components/App.tsx]: Done

File: Components/App.tsx (hunk 6 above)
Comment (fouadkachlan) [reply Components/App.tsx]: Done

File: Components/App.tsx (hunk 4 above)
Comment (fouadkachlan) [reply Components/App.tsx]: Done

File: Components/App.tsx (hunk 6 above)
Comment (jihad292) [reply Components/App.tsx]: it is not done bro ba3da mawjoude barat l CustomDiv l asese. ye3ne what i mean CustomDiv should have basicStyles in and also having as prop styles so the finalizedStyle will be {...basicStye,style}

Comment (fouadkachlan) [reply Components/App.tsx]: Done

File: Components/App.tsx [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+              width: '100%',
+              height: '40px',
+              borderRadius: '5%',
+              paddingLeft: '10px',
+              fontSize: '1.5vw',
+              border: '1px solid grey'
+            }} />
+          </CustomDiv>
+          <CustomButton onClick={addTask} style={
+            finalizedStyle({
```
Comment (jihad292) [inline Components/App.tsx]: no bro we have set fibalizedStyle inside style ye3ne ma ka2ana 3melna chi.

File: Components/CustomDiv/CustomDiv.tsx [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+interface CustomeDivProps {
```
Comment (jihad292) [inline Components/CustomDiv/CustomDiv.tsx:3]: const type  CustomeDivProps {
  children: React.ReactNode;
  style : React.CSSProperties;
width:number;
height:number;
display:"flex",
alignItems:"center"|"flex-start"|"flex-end",
justifyContent:"center"|"flex-start"|"flex-end"
}

File: Components/CustomDiv/CustomDiv.tsx [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+  children: React.ReactNode;
+  style : React.CSSProperties;
+}
+
+const CustomDiv: React.FC<CustomeDivProps> = ({ children, style }) => {
```
Comment (jihad292) [inline Components/CustomDiv/CustomDiv.tsx]: const basicstyles:CSSProperties = {
width,
height,
display,
alignItems,
justifyContent"}

Comment (jihad292) [reply Components/CustomDiv/CustomDiv.tsx]: const fanilizedStyles = {...basicStyles,style}

File: Components/App.tsx [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+    setDeadline(0);
+  };
+
+  const completeTask = (taskNameToDelete: string): void => {
+    setTodoList(todoList.filter((task) => {
+      return task.taskName !== taskNameToDelete;
+    }));
+  };
+
+ //const CustomDiv: React.FC<CustomeDivProps> = ({ children, style , width , height , display , alignItems , justifyContent  }) => {
```
Comment (jihad292) [inline Components/App.tsx]: remove unused code

File: Components/App.tsx [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+  return (
+    <CustomDiv width="100%" height="150%"  display="flex"  justifyContent="center" alignItems="center"  style={{ backgroundColor: '#f0f0f0'}}>
```
Comment (jihad292) [inline Components/App.tsx]: height 150% ?

File: Components/App.tsx [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+      <CustomDiv display="flex" justifyContent="center" alignItems="center" width="" height=""
```
Comment (jihad292) [inline Components/App.tsx]: always it is display "flex". so why we pass it as a prop?

File: Components/CustomDiv/CustomDiv.tsx [hunk 13]
```diff
@@ -0,0 +1,30 @@
+import React, { CSSProperties } from 'react';
+
+interface CustomeDivProps {
+  children: React.ReactNode;
+  style : React.CSSProperties;
+  width: number | string;
+  height: number | string;
```
Comment (jihad292) [inline Components/CustomDiv/CustomDiv.tsx]: why number or string ?

File: Components/App.tsx [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+          fontFamily: 'Arial, Helvetica, sans-serif',
+          backgroundColor: '#fff',
+          boxShadow: '0 0 10px rgba(0,0,0,0.1)',
+          borderRadius: '10px',
+          padding: '2%'
+        }
+      }>
+        <CustomDiv width="100%" height="80%" display="flex" alignItems="center" justifyContent="center"
+        style={
+            {flex: 30,
```
Comment (jihad292) [inline Components/App.tsx]: why we are using here flex:30?

File: Components/CustomDiv/CustomDiv.tsx (hunk 9 above)
Comment (fouadkachlan) [reply Components/CustomDiv/CustomDiv.tsx]: Done

File: Components/App.tsx (hunk 10 above)
Comment (fouadkachlan) [reply Components/App.tsx]: Done

File: Components/App.tsx (hunk 12 above)
Comment (fouadkachlan) [reply Components/App.tsx]: when I am removing display flex it is giving error that's why

File: Components/CustomDiv/CustomDiv.tsx (hunk 13 above)
Comment (fouadkachlan) [reply Components/CustomDiv/CustomDiv.tsx]: It should be string to accept % for example so I dropped the number
Done

File: Components/App.tsx (hunk 14 above)
Comment (fouadkachlan) [reply Components/App.tsx]: The flex: 30 property in my code seems to be an attempt to define the flex grow value, but it is not a standard or correct usage in css. In css, the flex property is a shorthand for flex-grow, flex-shrink, and flex-basis.

**Pull request closed without merging**